lib/pure/asyncdispatch.nim

```diff
@@ -368,15 +368,21 @@ proc all*[A](futs: seq[Future[A]]): Future[seq[A]] =
     completedFutures = 0
 
   for i, fut in futs:
-    fut.callback = proc(f: Future[A]) =
-      retValues[i] = f.read()
-      inc(completedFutures)
+    proc setCallback(i: int) =
+      fut.callback = proc(f: Future[A]) =
+        retValues[i] = f.read()
+        inc(completedFutures)
 
-      if completedFutures == len(futs):
-        retFuture.complete(retValues)
+        if completedFutures == len(retValues):
+          retFuture.complete(retValues)
+
+    setCallback(i)
 
   return retFuture
 
+proc all*[A](futs: varargs[Future[A]]): Future[seq[A]] =
+  return all(@futs)
```

Comment:
Well, I actually tried varargs first, but it doesn't work without conversion to seq, see this comment  https://github.com/nim-lang/Nim/pull/4241#issuecomment-222508429